### PR TITLE
Allow cancellation of buggy workflow

### PR DESCRIPTION
**Pull request by efischer19**
From https://openedx.atlassian.net/browse/TNL-6093 and https://splunk.edx.org/en-US/app/search/search?sid=1481833446.2325572, it's possible to run into errors that prevent delete student state. Stack trace:
```
Dec 15 18:59:22 ip-10-4-11-15 [service_variant=lms][django.request][env:prod-edge-edxapp] ERROR [ip-10-4-11-15  1730] [base.py:256] - Internal Server Error: /courses/course-v1:MITx+uINOV8x+4T2016/instructor/api/reset_student_attempts
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/newrelic-2.74.0.54/newrelic/hooks/framework_django.py", line 503, in wrapper
    return wrapped(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/decorators/http.py", line 45, in inner
    return func(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 110, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/decorators/cache.py", line 43, in _cache_controlled
    response = viewfunc(request, *args, **kw)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/instructor/views/api.py", line 202, in wrapped
    return func(*args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/instructor/views/api.py", line 175, in wrapped
    return func(*args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/instructor/views/api.py", line 126, in wrapped
    return func(request, *args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/instructor/views/api.py", line 1975, in reset_student_attempts
    delete_module=delete_module
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/instructor/enrollment.py", line 254, in reset_student_attempts
    requesting_user_id=requesting_user_id
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/openassessment/xblock/staff_area_mixin.py", line 514, in clear_student_state
    self._cancel_workflow(sub['uuid'], "Student state cleared", requesting_user_id=requesting_user_id)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/openassessment/xblock/staff_area_mixin.py", line 567, in _cancel_workflow
    assessment_requirements=assessment_requirements
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/openassessment/workflow/api.py", line 397, in cancel_workflow
    AssessmentWorkflow.cancel_workflow(submission_uuid, comments, cancelled_by_id, assessment_requirements)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/openassessment/workflow/models.py", line 552, in cancel_workflow
    workflow.cancel(assessment_requirements)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/openassessment/workflow/models.py", line 510, in cancel
    score = self.get_score(assessment_requirements, step_for_name)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/openassessment/workflow/models.py", line 261, in get_score
    score = get_score_func(self.submission_uuid, step_requirements)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/openassessment/assessment/api/peer.py", line 160, in get_score
    if not submitter_is_finished(submission_uuid, peer_requirements):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/openassessment/assessment/api/peer.py", line 63, in submitter_is_finished
    raise PeerAssessmentRequestError(u'Requirements dict must contain "must_grade" key')
PeerAssessmentRequestError: Requirements dict must contain "must_grade" key
```

These are just my first ideas of a fix that would allow the delete state to proceed.

### CODE REVIEW

**Comment by efischer19** [comment]
@cahrens @staubina Could you review this for potential inclusion in the upcoming release? The upshot of it is that we have a student who's workflow is stuck in `peer` state for a problem that doesn't have any peer steps. We'd like to fix that situation by deleting student state, but that hits the stacktrace listed above.

This was just my first though, if you have any other ideas of how to solve this problem don't be shy!

File: openassessment/workflow/models.py [hunk 1]
```diff
@@ -507,7 +508,10 @@ def cancel(self, assessment_requirements):
             if on_cancel_func is not None:
                 on_cancel_func(self.submission_uuid)
 
-        score = self.get_score(assessment_requirements, step_for_name)
+        try:
+            score = self.get_score(assessment_requirements, step_for_name)
+        except PeerAssessmentRequestError:
```
Comment (cahrens) [inline openassessment/workflow/models.py]: I would suggest a more general error, and add a logging statement in the except.

Comment (efischer19) [inline openassessment/workflow/models.py]: 👍 Thanks!

**Comment by staubina** [comment]
@cahrens Tests are passing please review before merge.

**Comment by cahrens** [comment]
👍

**Comment by staubina** [comment]
👍

**Pull request merged**